### DISCUSSION TITLE
[ideas] Could data.frame/tibble columns be branches?

### DISCUSSION THREAD

**Discussion by JVAQUEROM** [ropensci/targets#1328] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Sometimes I am working with a data.frame or tibble (it is a target) and I realize that I want to add a column to it. However, if I modify this target by adding a new column, all the pipeline that depends on it will be re-run, despite the fact that the columns have been unaffected (I only *created* a column, but did not modify the rest!). So I was wondering, since a data.frame/tible is just a list with a few tweaks (same length in each element, etc.), if it would be possible that each column was a branch and the pipeline could "know" what columns are accessed in each target and decide if it is necessary to re-run or not.

Maybe this is technically impossible but I wanted to share the idea.
Regards!

**Comment by wlandau** [comment]
`iteration = "list"` on the upstream data frame target should allow downstream targets to branch over the columns instead of the rows.

**Reply by JVAQUEROM** [reply]
Thanks for the suggestion, I'll make some tests with this, see how it works. Thanks!! Targets is great!
